tests/foreman/api/test_host.py

```diff
@@ -1349,3 +1349,39 @@ def test_positive_verify_files_with_pxegrub2_uefi_secureboot(self):
 
         @caselevel: System
         """
+
+    @tier1
+    def test_positive_read_puppet_proxy_name(self):
+        """Read a hostgroup created with puppet proxy and inspect server's
+        response
+
+        @id: 8825462e-f1dc-4054-b7fb-69c2b10722a2
+
+        @Assert: Field 'puppet_proxy_name' is returned
+
+        @BZ: 1371900
+        """
+        proxy = entities.SmartProxy().search(query={
+            'search': 'url = https://{0}:9090'.format(settings.server.hostname)
+        })[0]
+        host = entities.Host(puppet_proxy=proxy).create().read_json()
```

Comment:
@oshtaier Yes, I was expecting that question but unfortunately forgot to clarify this in PR message. Sorry for that.
You're absolutely right that `create` does perform `read`, however `create_json` doesn't, it's just return response for PUT method. However, BZ is about GET method. While responses should be identical, I've decided to stick to the BZ description.
While `read_json` is needed instead of `read`? Because BZ is about `*_name` fields, which are ignored by `read` and can be obtained by inspecting direct json response instead of nailgun object.
```python
 def create(self, create_missing=None):
    ...
    return self.read(attrs=self.create_json(create_missing))
```
```python
def create_json(self, create_missing=None):
    response = self.create_raw(create_missing)
    response.raise_for_status()
    return response.json()
```